lib/config/system.config.yml

```diff
@@ -8,6 +8,9 @@ db:
 cli:
   url: http://localhost:9876
 
+plugins:
+  # eg-example-plugin:
```

Comment:
`express-gateway-plugin-example` would be the package name, the actual plugin name would be `example` in the `system.config.yml` ?

for example:
`express-gateway-plugin-oauth2` would be the npm module the entry  within `system.config.yml` - `oauth2`

Reply:
ok, make sense. and if user needs to install non convention package they will have 'package' word for that?

Reply:
This section basically acts as a whitelist, yeah?

If so, why do we need a whitelist if the package has already been explicitly installed?

Reply:
I like concept installed vs enabled. Same idea as behind policies - have many registered(installed), but use only whitelisted(enabled). 
It seems to be standard pattern for plugins. Because installation is typically complex process and enable\disable is 1 sec operation.

Atom\VS Code\Sublime all have that. Kong - same idea.

Reply:
Even ExpressJS - installed middleware, unless it is used in code doesn't exist

Reply:
Ah, right.  So it's not really a "whitelist" of allowed plugins; it's just a signal to enable.  We could have discovery logic for finding a package listed here.

1. Try wrapping the package name in `express-gateway-${name}-plugin`.
2. If that fails, try `${name}`.
3. Error.

Reply:
At this point I've implemented the logic 
read settings for ${name} 
if contains `package` -> require('package')
else  ensure  `name` is prefixed with `express-gateway-plugin-` -> require(name)

So if you really want different name, np use `package` property

Reply:
Got it.  Thanks.

Reply:
👍  sounds good to me, the `plugins:` is so that there is a known declaration and exposure to what global params are in place 

simliarly in `policies:` you can have policies installed but unless you declare it, it can't be referenced in a pipeline